lib/evaluate/fold.cc

```diff
@@ -755,10 +825,14 @@ Expr<Type<TypeCategory::Real, KIND>> FoldOperation(FoldingContext &context,
       if (auto *expr{args[0].value().UnwrapExpr()}) {
         return ToReal<KIND>(context, std::move(*expr));
       }
+    } else if (name == "reshape") {
```

Comment:
We're going to want to find a way to fold the transformational intrinsic functions without having to duplicate the code for six different type categories for each one.

Reply:
Sure, what about a single entry point for `FunctionRef` that catches shared functions and then calls type specializations ? I have done this.
I also quickly checked if we could add some other shared path (e.g `FoldFloatinPointsIntrinsicFunctions`), but I think it would not save a lot of lines and it would makes the whole thing harder to follow.